src/Utilities/sharedFunctions.js

```diff
@@ -4,16 +4,29 @@ export const subtractWeeks = (numOfWeeks, date = new Date()) => {
     return date;
 };
 
-export const verifyDisconnectedSystem = (perReporterStaleness) => {
-    if (perReporterStaleness && !perReporterStaleness.puptoo) {
-        return true;
-    } else if (perReporterStaleness && perReporterStaleness.puptoo) {
-        const stalenessDate = new Date(perReporterStaleness.puptoo.stale_timestamp);
-        const twoWeeksPeriod = subtractWeeks(2);
-        if (twoWeeksPeriod > stalenessDate) {
-            return true;
-        }
+export const verifyCollectorStaleness = (reporterStaleness) =>{
+    const stalenessDate = new Date(reporterStaleness.stale_timestamp);
+    const currentDateTime = new Date();
+
+    const twoWeeksPeriod = subtractWeeks(2);
+    const oneWeeksPeriod = subtractWeeks(1);
+
+    if (stalenessDate > currentDateTime) {
```

Comment:
This is more of a question/nitpick. Do you think its more readable if you make the conditionals something like this?

```suggestion
}; if (stalenessDate > currentDateTime) {
        return 'Fresh';
    } else if ((oneWeeksPeriod < stalenessDate) && (stalenessDate < currentDateTime)) {
        return 'Stale';
    }
    else if ((twoWeeksPeriod < stalenessDate) && (stalenessDate < oneWeeksPeriod)) {
        return 'Stale warning';
    } else {
        return 'Culled';
    }
```

Reply:
You can find an insights-client not connected system by filtering with the 'Insights not connected' data collector. It is among the filters in the inventory table toolbar. I admit that this if/else block is not optimal. But this condition can not be achieved with other ways in my opinion (switch...).

Reply:
Oh the if/else chain I posted didnt change it, it just adds parentheses inside the conditional.